### PR TITLE
repo: set home dir when importing keys to show

### PR DESCRIPTION
**Pull request by tigarmo**
Unlike `--show-keys`, `--import` with `--import-options show-only` needs a directory to place the temporarily imported key; If not provided, gpg will default to ~/.gnupg, creating it if necessary. This creation can be a race when running tests in parallel, as multiple gpg processes try to create ~/.gnupg simultaneously.

- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

File: craft_archives/repo/apt_key_manager.py [hunk 1]
```diff
@@ -267,3 +272,12 @@ def _create_keyrings_path(self):
                 f"Keyrings location {self._keyrings_path} doesn't exist; Attempting to create it."
             )
             self._keyrings_path.mkdir(mode=0o755, parents=False)
+
+
+@contextmanager
+def _temporary_home_dir() -> Iterator[pathlib.Path]:
```
Comment (tigarmo) [inline craft_archives/repo/apt_key_manager.py:278]: not sure if `Iterator[Path]` is the correct annotation here

Comment (lengau) [reply craft_archives/repo/apt_key_manager.py:278]: Maybe `typing.ContextManager`?

**Review by lengau — APPROVED** [review]
Thanks for this! I'm happy with whatever you decide about the return type comment.

File: craft_archives/repo/apt_key_manager.py (hunk 1 above)
Comment (tigarmo) [reply craft_archives/repo/apt_key_manager.py:278]: Oh neat I tried `ContextManager` and pyright complained that the thing is actually an `Iterator`, so I guessed right after all

**Pull request merged**